packages/dapp-connector/src/WalletApi/Cip30Wallet.ts

```diff
@@ -50,6 +109,16 @@ export class Cip30Wallet {
     this.#authenticator = authenticator;
   }
 
+  /**
+   * Receives the array of extensions provided when `wallet.enable` was called and
+   * returns a list of methods names for the CIP-30 API and supported extensions.
+   */
+  #getAllowedApiMethods(extensions: WalletApiExtension[] = []): WalletMethod[] {
+    const enabledExtensions: WalletApiExtension[] =
```

Comment:
we could have a try/catch block here to account for invalid extensions param format + a test for it.
```diff
+ try {
    const enabledExtensions: WalletApiExtension[] =
      extensions.filter((extension) => this.supportedExtensions.some(({ cip }) => cip === extension.cip)) ?? [];
+ } catch(error) {
+     throw new ApiError(APIErrorCode.InvalidRequest, `Couldn't parse extensions ${extensions}.`);
+ }
```

Reply:
Added here c1977cc153c7fed6618328c611c363f02c8053f4, but this still enables the API as it fails after a successful `requestAccess`. Is this ok? Or should we validate the extensions array format before calling `requestAccess`?

Reply:
Nice catch 🕵️ ! I think it would be best to fail, so validate it before calling `requestAccess`.

Reply:
Imho we should throw if extensions don't match the type. If we do this, then devs are going to try it without reading the cip, find that it works, then use it in an invalid way. Other wallets won't thank us for it :sweat_smile:

Reply:
Then I think this should do it 5a764cd6110f7571bc3b0a96ae7a0e997e585ad7